js/main.js

```diff
@@ -18,83 +67,49 @@ function localStorageAv() {
   }
 }
 
-const storageAvailability = localStorageAv();
-
 function HandleInputData() {
-  const jsonData = JSON.stringify(booksArray);
-  localStorage.setItem('books', jsonData);
+  const jsonData = JSON.stringify(books.booksData);
+  localStorage.setItem('data', jsonData);
 }
 
-const CheckInput = () => { // eslint-disable-line no-unused-vars
+const storageAvailability = localStorageAv();
+
+const CheckInput = () => {
   if (storageAvailability) {
     HandleInputData();
   }
 };
 
-const removeBook = function (div) {
-  const removeBttn = document.getElementById(div.id).querySelector('.removeButton');
-  removeBttn.addEventListener('click', () => {
-    booksArray.splice(div.id, 1);
-    CheckInput();
-    div.remove();
-  });
-};
-
-const insertBookStructure = function (book, initialDisplay) {
-  const divSection = document.createElement('div');
-  divSection.innerHTML = `<p>${book.title}</p>`
-    + `<p>${book.author}</p>`
-    + '<button class = \'removeButton\'>Remove</button><br>';
-  if (!initialDisplay) {
-    divSection.id = `${booksArray.length - 1}`;
-  } else {
-    divSection.id = `${ids}`;
-    ids += 1;
+function hasValue(input) {
+  if (input === '') {
+    return false;
   }
-
-  booksDisplaySection.appendChild(divSection);
-  removeBook(divSection);
-};
-
-const createBook = function (titleValue, authorValue) {
-  const newBook = new Book(titleValue, authorValue);
-  booksArray.push(newBook);
-  insertBookStructure(newBook, false);
-};
+  return true;
+}
 
 form.addEventListener('click', (event) => {
   const inputs = document.querySelectorAll("#bookForm input[type='text']");
   const titleValue = inputs[0].value;
   const authorValue = inputs[1].value;
-  createBook(titleValue, authorValue);
-  CheckInput();
-  event.preventDefault();
-});
-
-const initializeBooks = function (data) {
-  if (data.length > 0) {
-    booksArray = data;
-  }
-};
-
-function hasValue(input) {
-  if (input === '') {
-    return false;
+  if (hasValue(titleValue) && hasValue(authorValue)) {
+    const book = new Book(titleValue, authorValue);
+    Methods.createBook(book);
+    CheckInput();
+    event.preventDefault();
   }
-  return true;
-}
+});
 
 function CheckLocalInput() {
-  const data = JSON.parse(localStorage.getItem('books'));
+  const data = JSON.parse(localStorage.getItem('data'));
   if (data !== null) {
-    initializeBooks(data);
-    ids = 0;
+    books.initializeBooks(data);
     for (let i = 0; i < data.length; i += 1) {
       if (hasValue(data[i])) {
-        insertBookStructure(data[i], true);
+        Methods.insertBookStructure(data[i], true);
       }
     }
   }
 }
 
 CheckLocalInput();
+/* eslint max-classes-per-file: off */
```

Comment:
- [ ] Make sure you are using ES6 syntax. Please go ahead and have a look [here](https://www.w3schools.com/js/js_es6.asp). 
- [ ] It uses JavaScript classes instead of objects and arrays, please note you can use objects and arrays but it has to be inside the classes.
```
class Books {
  constructor() {
    this.books = [];
  }

  addBook(title, author) {
    id = generateID(); // Optional
    this.books = this.books.concat({ title, author, author });
    return this.books; 
  }

  removeBook(bookID) {

    // bookID can be any other property, e.g. title or author, in as much is it unique
    this.books = this.books.filter(({ id }) => bookID !== id); // This assumes that `bookID` parameter is the book id
    return this.books; 
  }

  // getter method
  getBooks() {
    return this.books;
  }
}
```

- [ ] Make sure it has a class method to remove books.

Reply:
Hey @arslanbisharat, we have some doubts about your requirements :confused: Can you please expand? We have three different classes and are using objects and arrays inside of them. We have the Methods class with add and remove methods as the requirements described. We use the max-classes-per-file: off in line 15 to keep one single js file. Please help us understand what you are asking!

Reply:
Hi @mateo951, 
Thank you for asking me this question. I think you Don't need different classes for each function. 
About remove function, I really apologize for it that I missed it. For the next reviewer please ignore that requirement as it's already met.